### PR TITLE
Modified Github Actions workflows, removed check_repo  and unified if syntax

### PR DESCRIPTION
**Pull request by TianlongLiang**

### CODE REVIEW

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 1]
```diff
@@ -59,61 +59,23 @@ jobs:
...
         os: [ubuntu-20.04, ubuntu-22.04]
         include:
           - os: ubuntu-20.04
-            light: ${{ needs.check_repo.outputs.traffic_light_on_ubuntu_2004 }}
           - os: ubuntu-22.04
-            light: ${{ needs.check_repo.outputs.traffic_light_on_ubuntu_2204 }}
     steps:
-      - name: light status
-        run: echo "matrix.os=${{ matrix.os }}, light=${{ matrix.light }}"
+      - name: os status
```
[the remark this answers is not in the document]
Comment (TianlongLiang) [reply .github/workflows/compilation_on_android_ubuntu.yml]: done

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 2]
```diff
@@ -492,8 +414,8 @@ jobs:
         working-directory: ./tests/wamr-test-suites
 
   spec_test_extra:
-    if: ${{ endsWith(github.repository, 'wasm-micro-runtime') }}
-    needs: [build_iwasm, build_llvm_libraries, build_wamrc, check_repo]
+    if: endsWith(github.repository, 'wasm-micro-runtime')
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_android_ubuntu.yml:417]: This `if` is not necessary, too

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 3]
```diff
@@ -532,8 +454,8 @@ jobs:
         working-directory: ./tests/wamr-test-suites
 
   spec_test_x86_32:
-    if: ${{ endsWith(github.repository, 'wasm-micro-runtime') }}
-    needs: [build_iwasm, build_llvm_libraries, build_wamrc, check_repo]
+    if: endsWith(github.repository, 'wasm-micro-runtime')
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_android_ubuntu.yml:457]: This `if` is not necessary, too

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 4]
```diff
@@ -126,32 +85,25 @@ jobs:
...
         working-directory: build-scripts
 
   build_wamrc:
-    needs: [build_llvm_libraries, check_repo]
+    needs: [build_llvm_libraries]
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-20.04, ubuntu-22.04]
         include:
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_android_ubuntu.yml:98]: `include` should be removed, too.

File: .github/workflows/compilation_on_android_ubuntu.yml [hunk 5]
```diff
@@ -249,21 +200,15 @@ jobs:
             make_options_feature: "-DWAMR_BUILD_MINI_LOADER=1"
         include:
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_android_ubuntu.yml:201]: `include` should be removed

File: .github/workflows/compilation_on_macos.yml [hunk 6]
```diff
@@ -51,47 +51,19 @@ jobs:
...
-            echo "::set-output name=light::red"
-          fi
-
   build_llvm_libraries:
-    needs: check_repo
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [macos-latest]
         include:
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_macos.yml:59]: remove `include`

File: .github/workflows/compilation_on_macos.yml [hunk 7]
```diff
@@ -104,30 +76,24 @@ jobs:
...
 
   build_wamrc:
-    needs: [build_llvm_libraries, check_repo]
+    needs: [build_llvm_libraries]
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [macos-latest]
         include:
           - os: macos-latest
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_macos.yml:90]: remove `include`

File: .github/workflows/compilation_on_macos.yml [hunk 8]
```diff
@@ -226,19 +191,14 @@ jobs:
             make_options_feature: "-DWAMR_BUILD_MINI_LOADER=1"
         include:
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_macos.yml:192]: remove `include`

File: .github/workflows/compilation_on_sgx.yml [hunk 9]
```diff
@@ -51,47 +52,19 @@ jobs:
...
-            echo "::set-output name=light::green"
-          fi
-
   build_llvm_libraries:
-    needs: check_repo
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-20.04]
         include:
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_sgx.yml:60]: remove `include`

File: .github/workflows/compilation_on_sgx.yml [hunk 10]
```diff
@@ -154,13 +126,8 @@ jobs:
             make_options_feature: "-DWAMR_BUILD_MINI_LOADER=1"
         include:
           - os: ubuntu-20.04
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_sgx.yml:128]: remove `include`

File: .github/workflows/compilation_on_sgx.yml [hunk 11]
```diff
@@ -174,32 +141,25 @@ jobs:
...
 
   build_wamrc:
-    needs: [build_llvm_libraries, check_repo]
+    needs: [build_llvm_libraries]
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         os: [ubuntu-20.04]
         include:
           - os: ubuntu-20.04
```
Comment (lum1n0us) [inline .github/workflows/compilation_on_sgx.yml:160]: remove `include`

**Review by lum1n0us — APPROVED** [review]
LGTM

**Pull request merged**